lib/krane/cluster_resource_discovery.rb

```diff
@@ -17,13 +17,38 @@ def crds
     end
 
     def global_resource_kinds
-      @globals ||= fetch_globals.map { |g| g["kind"] }
+      @globals ||= fetch_resources(only_globals: true).map { |g| g["kind"] }
+    end
+
+    def prunable_resources
+      api_versions = fetch_api_versions
+      fetch_resources.map do |resource|
+        next unless resource['verbs'].include?("delete")
+        version = api_versions.fetch(resource['apigroup'], ['v1']).last
+        [resource['apigroup'], version, resource['kind']].compact.join("/")
+      end.compact
     end
 
     private
 
-    def fetch_globals
-      raw, _, st = kubectl.run("api-resources", "--namespaced=false", output: "wide", attempts: 5,
+    def fetch_api_versions
```

Comment:
Could you add a comment documenting an example of the format/fields returned by this and `fetch_resources`? I think it would make it easier to review this class.